frontend/src/routes/Credentials/CredentialsForm.tsx

```diff
@@ -1360,8 +1420,48 @@ export function CredentialsForm(
       if (isEditing) {
         const secret = credentialData as Secret
         const patch: { op: 'replace'; path: string; value: unknown }[] = []
+
         if (secret.stringData) {
-          patch.push({ op: 'replace', path: `/stringData`, value: secret.stringData })
+          let client_id = ''
+          let client_secret = ''
+          let ocmClusterManagerOptions = ''
+          let ocmAPIToken = ''
+
+          // In Editing mode, extract the new values from the form `formData` and merge them with the existing `stringData` object
+          formData.sections.forEach((section: Section | SectionGroup) => {
+            if (section.title === 'OpenShift Cluster Manager' && 'inputs' in section) {
+              section.inputs?.forEach((input: Input) => {
+                if (typeof input.value === 'string') {
+                  switch (input.id) {
+                    case 'client_id':
+                      client_id = input.value
+                      break
+                    case 'client_secret':
+                      client_secret = input.value
+                      break
+                    case 'ocmClusterManagerOptions':
+                      ocmClusterManagerOptions = input.value
+                      break
+                    case 'ocmAPIToken':
+                      ocmAPIToken = input.value
+                      break
+                    default:
+                      break
+                  }
+                }
+              })
+            }
+          })
+          // Merge existing stringData with new values
+          const updatedStringData = {
+            ...secret.stringData,
+            // Add or overwrite with new values from the form
+            client_id,
+            client_secret,
+            ocmClusterManagerOptions,
+            ocmAPIToken,
+          }
+          patch.push({ op: 'replace', path: `/stringData`, value: updatedStringData })
```

Comment:
This fix is very specific to this type of credential. If you look at the underlying secret, there are a couple issues:

- The inapplicable values are not deleted from the secret, but they are set to empty string.
- We accidentally added the key `ocmClusterManagerOptions`
- These 4 fields get added to other credential types when you edit them

![image](https://github.com/user-attachments/assets/a05150b6-112f-48f8-8d3f-a4d18b83cfd1)

This was the type of fix I was suggesting when I mentioned converting the values to base64 and using `data` rather than `stringData`. This will make sure the object is fully replaced. Please play around with it, take a look at the Secrets in OpenShift and verify they are being updated correctly.

```suggestion
          const data: Secret['data'] = {}
          Object.keys(secret.stringData).forEach((key) => {
            if (secret.stringData?.[key]) {
              data[key] = Buffer.from(secret.stringData[key], 'ascii').toString('base64')
            }
          })
          patch.push({ op: 'replace', path: `/data`, value: data })
```

Reply:
@KevinFCormier Good catch, I missed checking that! Thanks for reviewing and providing feedback. I'm updating the code with the logic you suggested.